src/Engine/EmitMSIL/Operators.cs

```diff
@@ -0,0 +1,244 @@
+        {
+            bool lhs = ToBoolean(a);
+            bool rhs = ToBoolean(b);
+
+            return lhs || rhs;
+        }
+
+        public static bool Not(dynamic a)
+        {
+            bool data = ToBoolean(a);
+
+            return !data;
+        }
+        public static bool Eq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab == bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return MathUtils.Equals(ad, bd);
+                }
+                return a == b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) == 0;
+            }
+
+            return a.Equals(b);
+        }
+
+        public static bool Nq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab != bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return !MathUtils.Equals(ad, bd);
+                }
+                return a != b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) != 0;
+            }
+
+            return !a.Equals(b);
+        }
+
+        public static bool Gt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a > b;
+            }
+
+            return false;
+        }
+
+        public static bool Lt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a < b;
+            }
+
+            return false;
+        }
+
+        public static bool Ge(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a >= b;
+            }
+
+            return false;
+        }
+
+        public static bool Le(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a <= b;
+            }
+
+            return false;
+        }
+
+        public static bool IsOperator(string name)
+        {
+            if (name.Length > 0)
```

Comment:
same here, I think you can replace this with checks on the AST nodes directly instead of doing string comparisons.